### PR TITLE
Test screenshot on failure

### PR DESCRIPTION
**Pull request by hybrist**
Copied the file(s) from testium & translated to javascript.
- `test/integration`: Tests that launch an app and successfully run against it
- `test/*.js`: Tests that potentially spawn mocha and should not run in the same process as `test/integration` (because of port conflicts)
- `test/screenshot_integration`: A failing test suite to test screenshot taking

### CODE REVIEW

File: test/mocha.opts [hunk 1]
```diff
@@ -1,3 +1,2 @@
 --compilers js:babel-core/register
---recursive
```
Comment (hybrist) [inline test/mocha.opts:2]: That way nested directories, e.g. `test/screenshot_integration`, are not included in `mocha` which is what we want.

File: test/screenshot_integration/force_screenshot.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+describe('forced screenshot', () => {
+  before(browser.beforeHook());
+
+  it('my test', () => {
+    this.browser.navigateTo('/');
+    // This is supposed to be failing, the real status code is 200
+    assert.equal('statuscode', 418, browser.getStatusCode());
+  });
+
+  it('some !%#(*.>:; sPecial  chars', () => {
```
Comment (ghost) [inline test/screenshot_integration/force_screenshot.test.js]: Might want to include an underscore in there

File: test/screenshots.test.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+const SCREENSHOT_DIRECTORY = `${__dirname}/screenshot_integration_log/screenshots`;
+const TEST_FILE = 'test/screenshot_integration/force_screenshot.test.js';
+
+const ENV_OVERRIDES = {
+  testium_logDirectory: LOG_DIRECTORY,
+  testium_screenshotDirectory: SCREENSHOT_DIRECTORY,
+};
+
+describe('screenshots', () => {
+  before(`rm -rf ${LOG_DIRECTORY}`, (done) =>
```
Comment (ghost) [inline test/screenshots.test.js]: No brackets needed for `(done) =>`

**Comment by ghost** [comment]
Looks good otherwise :thumbsup: :shipit:

File: test/screenshot_integration/force_screenshot.test.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { browser } from '../../';
+import assert from 'assertive';
+
+describe('forced screenshot', () => {
+  before(browser.beforeHook());
+
+  it('my test', () => {
+    this.browser.navigateTo('/');
```
Comment (EndangeredMassa) [inline test/screenshot_integration/force_screenshot.test.js:8]: this test fails here instead of 2 lines below because it shouldn't reference `this`

Comment (hybrist) [inline test/screenshot_integration/force_screenshot.test.js:8]: Nice catch! Fixing in a follow-up.

**Pull request merged**